### ISSUE TITLE
Address probe timeout is too short

### ISSUE THREAD

**Issue by yadudoc** [Parsl/parsl#1666]
**Describe the bug**
When running 600 nodes on summit and 800 nodes on theta, about 10% of the managers failed to register with the Interchange. Instead they failed with: 
```
2020-04-30 19:46:12.348 parsl:127 Rank:0 [ERROR]  Caught exception while trying to determine viable address to interchange
Traceback (most recent call last):
  File "/dev/shm/candle_inferpy3.7/bin/process_worker_pool.py", line 119, in __init__
    raise Exception("No viable address found")
Exception: No viable address found
```

**To Reproduce**
Run parsl on theta with ~800 nodes

**Expected behavior**
We know that managers are definitely able to connect. It is okay for some managers to take more time to register.

**Environment**
 - OS: linux
 - Python 3.7
 - Parsl version: master

**Comment by yadudoc** [comment]
Based on @benclifford 's recommendation that the timeout should be configurable, I propose the following configurable option, with an updated default of 30s instead of the current 2s : 
```
HighThroughputExecutor(address_probe_timeout = 30)
```

**Comment by benclifford** [comment]
From that error message, it isn't clear to me that this really is a timeout problem (vs. cannot connect in some other way), but I think the timeout work in #1667 is a good thing anyway.

**Comment by yadudoc** [comment]
Closed by #1667
